### PR TITLE
fix Get schema by version can get the deleted schema info #6754

### LINKED ISSUE #6754: Get schema by version can get the deleted schema info.
**Describe the bug**
Get schema by version can get the deleted schema info. 

**To Reproduce**
1. create a schema for a topic.
2. delete schema of the topic.
3. get schema by the previous schema version that created.
4. Still can schema info.

**Expected behavior**
Should response 404

**Additional context**
master branch.

### PR DESCRIPTION
**Pull request by 315157973**
fix #6754

### CODE REVIEW

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/service/PersistentTopicE2ETest.java [hunk 1]
```diff
@@ -83,6 +53,16 @@
 import org.testng.annotations.DataProvider;
 import org.testng.annotations.Test;
 
+import java.lang.reflect.Field;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.*;
```
Comment (jiazhai) [inline pulsar-broker/src/test/java/org/apache/pulsar/broker/service/PersistentTopicE2ETest.java:62]: .* should be avoid

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/service/PersistentTopicE2ETest.java [hunk 2]
```diff
@@ -83,6 +53,16 @@
...
+
+import static org.testng.Assert.*;
```
Comment (jiazhai) [inline pulsar-broker/src/test/java/org/apache/pulsar/broker/service/PersistentTopicE2ETest.java:64]: .* should be avoid

**Comment by jiazhai** [comment]
Thanks for the work. @codelipenghui  to take a look.

**Comment by 315157973** [comment]
> We don't need to delete the ledger that stores schemas, the schema delete just add a delete mark in the ledger. There is a method to trim the deleted schemas int the SchemaRegistryServiceImpl. Maybe you can use it directly.
> https://github.com/apache/pulsar/blob/9d3ab60efddd3f4064789293fd8e9c4d4388b6c7/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/schema/SchemaRegistryServiceImpl.java#L317

Thanks,"trimDeletedSchemaAndGetList" is mainly used to get all the schemas and filter out the deleted ones. It is a query method, not really trimming the schema. 
If there is no need to delete Ledger, then I will remove this piece of code.

**Comment by sijie** [comment]
@codelipenghui can you review this again?

**Comment by codelipenghui** [comment]
@315157973 Do we need to change the schema delete logic for fixing the issue? I don't see what's the problem with the current deletion logic, just see the get schema version does not filter out the deleted schema. The delete schema method is unused and I think the current logic uses `deleteSchema` method in the `SchemaRegistryServiceImpl`. If I missed something, please point out, thanks.

**Comment by 315157973** [comment]
> @315157973 Do we need to change the schema delete logic for fixing the issue? I don't see what's the problem with the current deletion logic, just see the get schema version does not filter out the deleted schema. The delete schema method is unused and I think the current logic uses `deleteSchema` method in the `SchemaRegistryServiceImpl`. If I missed something, please point out, thanks.

Thanks for correcting me，filtering is a better solution

**Pull request merged**